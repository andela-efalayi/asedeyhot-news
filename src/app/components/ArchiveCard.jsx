import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardText } from 'material-ui/Card';

const ArchiveCard = ({ headlines }) => {
  if (Object.keys(headlines).length === 0) {
    return (
      <div>
        <h4>You have no saved headlines.</h4>
      </div>
    );
  }
  return (
    <div>
      {Object.entries(headlines).map(([key, value]) => (
      <Card key={key} className="archiveItem">
        <div className="row">
          <div className="col-xs-4 col-sm-4">
            <img className="img-responsive" src={value.urlToImage} />
          </div>
          <div className="col-xs-8 col-sm-8">
            <CardText>
              <h5>{value.title}</h5>
              <p>Author: {value.author}</p>
              <p><a href={value.url}>Go to link</a></p>
            </CardText>
          </div>
        </div>
      </Card>
    ))}
    </div>
  );
};

ArchiveCard.propTypes = {
  headlines: PropTypes.object
};

ArchiveCard.defaultProps = {
  headlines: null
};

export default ArchiveCard;
