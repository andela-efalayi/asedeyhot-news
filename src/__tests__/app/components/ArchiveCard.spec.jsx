import React from 'react';
import { shallow } from 'enzyme';
import ArchiveCard from
'../../../app/components/ArchiveCard.jsx';

const mockData = {
  author: 'BBC News',
  title: 'General Election 2017: Labour to rip up Tory Brexit plan',
  description: 'The party says it would prioritise jobs.',
  url: 'http://www.bbc.co.uk/news/uk-politics-39698465',
  urlToImage: 'https://ichef-1.bbci.co.uk/news/12509/95771057_mediaitem95771056.jpg',
  publishedAt: '2017-04-25T06:32:28+00:00'
};

describe('Favourites Component', () => {
  it('should render Favourites', () => {
    const wrapper = shallow(
        <ArchiveCard headlines={mockData}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
