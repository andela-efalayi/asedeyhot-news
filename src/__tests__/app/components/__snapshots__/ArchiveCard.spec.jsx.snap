// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites Component should render Favourites 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Card
        className="archiveItem"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <div
            className="row"
        >
            <div
                className="col-xs-4 col-sm-4"
            >
                <img
                    className="img-responsive"
                    src={undefined}
                />
            </div>
            <div
                className="col-xs-8 col-sm-8"
            >
                <CardText>
                    <h5 />
                    <p>
                        Author: 
                    </p>
                    <p>
                        <a
                            href={undefined}
                        >
                            Go to link
                        </a>
                    </p>
                </CardText>
            </div>
        </div>
    </Card>
    <Card
        className="archiveItem"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <div
            className="row"
        >
            <div
                className="col-xs-4 col-sm-4"
            >
                <img
                    className="img-responsive"
                    src={undefined}
                />
            </div>
            <div
                className="col-xs-8 col-sm-8"
            >
                <CardText>
                    <h5 />
                    <p>
                        Author: 
                    </p>
                    <p>
                        <a
                            href={undefined}
                        >
                            Go to link
                        </a>
                    </p>
                </CardText>
            </div>
        </div>
    </Card>
    <Card
        className="archiveItem"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <div
            className="row"
        >
            <div
                className="col-xs-4 col-sm-4"
            >
                <img
                    className="img-responsive"
                    src={undefined}
                />
            </div>
            <div
                className="col-xs-8 col-sm-8"
            >
                <CardText>
                    <h5 />
                    <p>
                        Author: 
                    </p>
                    <p>
                        <a
                            href={undefined}
                        >
                            Go to link
                        </a>
                    </p>
                </CardText>
            </div>
        </div>
    </Card>
    <Card
        className="archiveItem"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <div
            className="row"
        >
            <div
                className="col-xs-4 col-sm-4"
            >
                <img
                    className="img-responsive"
                    src={undefined}
                />
            </div>
            <div
                className="col-xs-8 col-sm-8"
            >
                <CardText>
                    <h5 />
                    <p>
                        Author: 
                    </p>
                    <p>
                        <a
                            href={undefined}
                        >
                            Go to link
                        </a>
                    </p>
                </CardText>
            </div>
        </div>
    </Card>
    <Card
        className="archiveItem"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <div
            className="row"
        >
            <div
                className="col-xs-4 col-sm-4"
            >
                <img
                    className="img-responsive"
                    src={undefined}
                />
            </div>
            <div
                className="col-xs-8 col-sm-8"
            >
                <CardText>
                    <h5 />
                    <p>
                        Author: 
                    </p>
                    <p>
                        <a
                            href={undefined}
                        >
                            Go to link
                        </a>
                    </p>
                </CardText>
            </div>
        </div>
    </Card>
    <Card
        className="archiveItem"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <div
            className="row"
        >
            <div
                className="col-xs-4 col-sm-4"
            >
                <img
                    className="img-responsive"
                    src={undefined}
                />
            </div>
            <div
                className="col-xs-8 col-sm-8"
            >
                <CardText>
                    <h5 />
                    <p>
                        Author: 
                    </p>
                    <p>
                        <a
                            href={undefined}
                        >
                            Go to link
                        </a>
                    </p>
                </CardText>
            </div>
        </div>
    </Card>
</div>,
  "nodes": Array [
    <div>
      <Card
            className="archiveItem"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-4 col-sm-4"
                  >
                        <img
                              className="img-responsive"
                              src={undefined}
                        />
                  </div>
                  <div
                        className="col-xs-8 col-sm-8"
                  >
                        <CardText>
                              <h5 />
                              <p>
                                    Author: 
                              </p>
                              <p>
                                    <a
                                          href={undefined}
                                    >
                                          Go to link
                                    </a>
                              </p>
                        </CardText>
                  </div>
            </div>
      </Card>
      <Card
            className="archiveItem"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-4 col-sm-4"
                  >
                        <img
                              className="img-responsive"
                              src={undefined}
                        />
                  </div>
                  <div
                        className="col-xs-8 col-sm-8"
                  >
                        <CardText>
                              <h5 />
                              <p>
                                    Author: 
                              </p>
                              <p>
                                    <a
                                          href={undefined}
                                    >
                                          Go to link
                                    </a>
                              </p>
                        </CardText>
                  </div>
            </div>
      </Card>
      <Card
            className="archiveItem"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-4 col-sm-4"
                  >
                        <img
                              className="img-responsive"
                              src={undefined}
                        />
                  </div>
                  <div
                        className="col-xs-8 col-sm-8"
                  >
                        <CardText>
                              <h5 />
                              <p>
                                    Author: 
                              </p>
                              <p>
                                    <a
                                          href={undefined}
                                    >
                                          Go to link
                                    </a>
                              </p>
                        </CardText>
                  </div>
            </div>
      </Card>
      <Card
            className="archiveItem"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-4 col-sm-4"
                  >
                        <img
                              className="img-responsive"
                              src={undefined}
                        />
                  </div>
                  <div
                        className="col-xs-8 col-sm-8"
                  >
                        <CardText>
                              <h5 />
                              <p>
                                    Author: 
                              </p>
                              <p>
                                    <a
                                          href={undefined}
                                    >
                                          Go to link
                                    </a>
                              </p>
                        </CardText>
                  </div>
            </div>
      </Card>
      <Card
            className="archiveItem"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-4 col-sm-4"
                  >
                        <img
                              className="img-responsive"
                              src={undefined}
                        />
                  </div>
                  <div
                        className="col-xs-8 col-sm-8"
                  >
                        <CardText>
                              <h5 />
                              <p>
                                    Author: 
                              </p>
                              <p>
                                    <a
                                          href={undefined}
                                    >
                                          Go to link
                                    </a>
                              </p>
                        </CardText>
                  </div>
            </div>
      </Card>
      <Card
            className="archiveItem"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-4 col-sm-4"
                  >
                        <img
                              className="img-responsive"
                              src={undefined}
                        />
                  </div>
                  <div
                        className="col-xs-8 col-sm-8"
                  >
                        <CardText>
                              <h5 />
                              <p>
                                    Author: 
                              </p>
                              <p>
                                    <a
                                          href={undefined}
                                    >
                                          Go to link
                                    </a>
                              </p>
                        </CardText>
                  </div>
            </div>
      </Card>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ArchiveCard
        headlines={
                Object {
                        "author": "BBC News",
                        "description": "The party says it would prioritise jobs.",
                        "publishedAt": "2017-04-25T06:32:28+00:00",
                        "title": "General Election 2017: Labour to rip up Tory Brexit plan",
                        "url": "http://www.bbc.co.uk/news/uk-politics-39698465",
                        "urlToImage": "https://ichef-1.bbci.co.uk/news/12509/95771057_mediaitem95771056.jpg",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "headlines": Object {
            "author": "BBC News",
            "description": "The party says it would prioritise jobs.",
            "publishedAt": "2017-04-25T06:32:28+00:00",
            "title": "General Election 2017: Labour to rip up Tory Brexit plan",
            "url": "http://www.bbc.co.uk/news/uk-politics-39698465",
            "urlToImage": "https://ichef-1.bbci.co.uk/news/12509/95771057_mediaitem95771056.jpg",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
          <Card
                    className="archiveItem"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-4 col-sm-4"
                              >
                                        <img
                                                  className="img-responsive"
                                                  src={undefined}
                                        />
                              </div>
                              <div
                                        className="col-xs-8 col-sm-8"
                              >
                                        <CardText>
                                                  <h5 />
                                                  <p>
                                                            Author: 
                                                  </p>
                                                  <p>
                                                            <a
                                                                      href={undefined}
                                                            >
                                                                      Go to link
                                                            </a>
                                                  </p>
                                        </CardText>
                              </div>
                    </div>
          </Card>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArchiveCard
    headlines={
        Object {
            "author": "BBC News",
            "description": "The party says it would prioritise jobs.",
            "publishedAt": "2017-04-25T06:32:28+00:00",
            "title": "General Election 2017: Labour to rip up Tory Brexit plan",
            "url": "http://www.bbc.co.uk/news/uk-politics-39698465",
            "urlToImage": "https://ichef-1.bbci.co.uk/news/12509/95771057_mediaitem95771056.jpg",
          }
    }
/>,
}
`;
